File: app/src/main/java/com/example/graduationprojectandroid/network/endpoints/APIs/UserAPI.java
package com.example.graduationprojectandroid.network.endpoints.APIs;
import com.example.graduationprojectandroid.network.UserData;
import com.google.gson.JsonObject;


import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface UserAPI {

    // for registration
    @POST("/is_login_unique")
    Call<String> isLoginUnique(
            @Query("login") String login
    );

    @POST("/is_password_ok")
    Call<String> isPasswordOK(
            @Query("password") String password
    );

    @POST("/register")
    Call<String> register(
            @Query("login") String login, @Query("password") String password
    );




    // log in
    @POST("/check_login_existence")
    Call<String> isLoginExists(
            @Query("login") String login
    );

    @POST("/enter")
    Call<String> enter(
            @Query("login") String login, @Query("password") String password
    );


    @GET("/check_token")
    Call<Boolean> checkToken(
            @Query("check_token") String token
    );







    @GET("/get_user_data")
    Call<UserData> getUserData(
            @Query("login") String login, @Query("token") String token
    );





    // for avatar
    @POST("/changed_avatar")
    Call<String> changedAvatar(@Body JsonObject jsonObject, @Query("token") String token);

    @GET("/check_avatar_name")
    Call<String> checkAvatarName(@Query("avatar_name") String avatarName);

}

File: app/src/main/java/com/example/graduationprojectandroid/network/endpoints/APIs/UserAPI.kt
package com.example.graduationprojectandroid.network.endpoints.APIs

import com.example.graduationprojectandroid.network.endpoints.ServerAnswer
import retrofit2.Call
import retrofit2.http.*
import java.util.ArrayList

interface UserAPI {

    @GET("mobileappusers/tryenter")
    fun test(
        @Query("login") login: String?,
        @Query("password") password: String?
    ): Call<String>

    @GET("mobileappusers/tryregister")
    fun tryRegister(
        @Query("login") login: String?,
        @Query("password") password: String?
    ): Call<ServerAnswer>

    @GET("mobileappusers/tryenter")
    fun tryEnter(
        @Query("login") login: String?,
        @Query("password") password: String?
    ): Call<ServerAnswer>

    @GET("mobileappusers/checktoken")
    fun checkToken(
        @Query("login") login: String?,
        @Query("token") token: String?
    ): Call<ServerAnswer>



    @GET("mobileappusers/getuserdata")
    fun getUserData(
        @Query("token") token: String?
    ): Call<ServerAnswer>

    // for avatar
    @GET("mobileappusers/changedavatar")
    fun changedAvatar(
        @Query("choosen_parts") choosenParts: ArrayList<Int?>?,
        @Query("avatar_name") avatarName: String?,
        @Query("token") token: String?
    ): Call<ServerAnswer>

    @GET("mobileappusers/checkavatarname")
    fun checkAvatarName(
        @Query("avatar_name") avatarName: String?
    ): Call<ServerAnswer>


    @GET("mobileappusers/getamountofonetypeavatarparts")
    fun getAmountOfOneTypeAvatarParts(
        @Query("avatar_part_type") partNumber: Int?
    ): Call<ServerAnswer>
}